fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/QuerySegmentAggregator.java

```diff
@@ -50,6 +50,9 @@
     private static final Set<String> SKIP_WHERE = new HashSet<>(Arrays.asList("_id"));
     
     protected Class<?> resourceType;
+    
+    // Used for whole system search on multiple resource types.
+    private List<String> resourceTypes = null;
```

Comment:
maybe set as a default Collections.emptyList();

Reply:
I prefer null here, because we will simply pass in one if need instead of add to existing list.

Reply:
right - however downstream you have two calls to check isNotNull and size 
it's purely preference.